<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="windowModality">
   <enum>Qt::ApplicationModal</enum>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1345</width>
    <height>798</height>
   </rect>
  </property>
  <property name="mouseTracking">
   <bool>true</bool>
  </property>
  <property name="acceptDrops">
   <bool>true</bool>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="layoutDirection">
   <enum>Qt::LeftToRight</enum>
  </property>
  <property name="autoFillBackground">
   <bool>true</bool>
  </property>
  <property name="documentMode">
   <bool>false</bool>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="QWidget" name="imageWidget" native="true">
      <property name="mouseTracking">
       <bool>true</bool>
      </property>
      <property name="acceptDrops">
       <bool>true</bool>
      </property>
      <property name="autoFillBackground">
       <bool>false</bool>
      </property>
      <widget class="QGroupBox" name="groupBox_3">
       <property name="geometry">
        <rect>
         <x>290</x>
         <y>470</y>
         <width>521</width>
         <height>141</height>
        </rect>
       </property>
       <property name="title">
        <string>命令调试助手</string>
       </property>
       <widget class="QSpinBox" name="constantTime">
        <property name="geometry">
         <rect>
          <x>280</x>
          <y>70</y>
          <width>58</width>
          <height>20</height>
         </rect>
        </property>
        <property name="maximum">
         <number>10000</number>
        </property>
        <property name="value">
         <number>1000</number>
        </property>
       </widget>
       <widget class="QCheckBox" name="constantSend">
        <property name="geometry">
         <rect>
          <x>210</x>
          <y>70</y>
          <width>71</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>连续发送</string>
        </property>
       </widget>
       <widget class="QCheckBox" name="showHex">
        <property name="geometry">
         <rect>
          <x>110</x>
          <y>70</y>
          <width>95</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>十六进制接收</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
       <widget class="QCheckBox" name="isHexSend">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>70</y>
          <width>95</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>十六进制发送</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
       <widget class="QPushButton" name="sendData">
        <property name="geometry">
         <rect>
          <x>170</x>
          <y>100</y>
          <width>75</width>
          <height>31</height>
         </rect>
        </property>
        <property name="text">
         <string>发送命令</string>
        </property>
       </widget>
       <widget class="QComboBox" name="sendContextBox">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>20</y>
          <width>411</width>
          <height>31</height>
         </rect>
        </property>
        <property name="editable">
         <bool>true</bool>
        </property>
       </widget>
       <widget class="QPushButton" name="saveCommand">
        <property name="geometry">
         <rect>
          <x>430</x>
          <y>20</y>
          <width>75</width>
          <height>31</height>
         </rect>
        </property>
        <property name="text">
         <string>保存指令</string>
        </property>
       </widget>
      </widget>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QDockWidget" name="functionDockWidget2">
      <property name="mouseTracking">
       <bool>true</bool>
      </property>
      <property name="acceptDrops">
       <bool>false</bool>
      </property>
      <property name="autoFillBackground">
       <bool>false</bool>
      </property>
      <property name="floating">
       <bool>false</bool>
      </property>
      <property name="windowTitle">
       <string>功能区</string>
      </property>
      <widget class="QWidget" name="dockWidgetContents_4">
       <widget class="QGroupBox" name="groupBox">
        <property name="geometry">
         <rect>
          <x>9</x>
          <y>0</y>
          <width>331</width>
          <height>743</height>
         </rect>
        </property>
        <property name="autoFillBackground">
         <bool>true</bool>
        </property>
        <property name="title">
         <string>功能区</string>
        </property>
        <widget class="QTabWidget" name="tabWidget">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>20</y>
           <width>311</width>
           <height>471</height>
          </rect>
         </property>
         <property name="currentIndex">
          <number>0</number>
         </property>
         <widget class="QWidget" name="tab">
          <attribute name="title">
           <string>串口设置</string>
          </attribute>
          <widget class="QGroupBox" name="groupBox_2">
           <property name="geometry">
            <rect>
             <x>30</x>
             <y>0</y>
             <width>251</width>
             <height>141</height>
            </rect>
           </property>
           <property name="title">
            <string/>
           </property>
           <widget class="QLabel" name="label_2">
            <property name="geometry">
             <rect>
              <x>0</x>
              <y>40</y>
              <width>42</width>
              <height>16</height>
             </rect>
            </property>
            <property name="text">
             <string>波特率:</string>
            </property>
           </widget>
           <widget class="QComboBox" name="stopBitCB">
            <property name="geometry">
             <rect>
              <x>170</x>
              <y>70</y>
              <width>71</width>
              <height>20</height>
             </rect>
            </property>
            <item>
             <property name="text">
              <string>1</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>1.5</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>2</string>
             </property>
            </item>
           </widget>
           <widget class="QComboBox" name="serialCB">
            <property name="geometry">
             <rect>
              <x>50</x>
              <y>10</y>
              <width>69</width>
              <height>20</height>
             </rect>
            </property>
           </widget>
           <widget class="QLabel" name="label">
            <property name="geometry">
             <rect>
              <x>0</x>
              <y>10</y>
              <width>51</width>
              <height>16</height>
             </rect>
            </property>
            <property name="text">
             <string>串口位:</string>
            </property>
           </widget>
           <widget class="QLabel" name="label_5">
            <property name="geometry">
             <rect>
              <x>130</x>
              <y>70</y>
              <width>42</width>
              <height>16</height>
             </rect>
            </property>
            <property name="text">
             <string>停止位:</string>
            </property>
           </widget>
           <widget class="QComboBox" name="baudRateCB">
            <property name="geometry">
             <rect>
              <x>50</x>
              <y>40</y>
              <width>71</width>
              <height>20</height>
             </rect>
            </property>
            <property name="currentText">
             <string>115200</string>
            </property>
            <property name="currentIndex">
             <number>11</number>
            </property>
            <item>
             <property name="text">
              <string>1200</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>1800</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>2400</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>4800</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>9600</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>14400</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>19200</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>38400</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>56000</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>57600</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>76800</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>115200</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>128000</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>256000</string>
             </property>
            </item>
           </widget>
           <widget class="QComboBox" name="checkCB">
            <property name="geometry">
             <rect>
              <x>50</x>
              <y>70</y>
              <width>71</width>
              <height>20</height>
             </rect>
            </property>
            <item>
             <property name="text">
              <string>empty</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>odd</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>even</string>
             </property>
            </item>
           </widget>
           <widget class="QLabel" name="label_4">
            <property name="geometry">
             <rect>
              <x>0</x>
              <y>70</y>
              <width>42</width>
              <height>16</height>
             </rect>
            </property>
            <property name="text">
             <string>校验位:</string>
            </property>
           </widget>
           <widget class="QComboBox" name="dataBitCB">
            <property name="geometry">
             <rect>
              <x>170</x>
              <y>40</y>
              <width>71</width>
              <height>20</height>
             </rect>
            </property>
            <property name="currentText">
             <string>5</string>
            </property>
            <item>
             <property name="text">
              <string>5</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>6</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>7</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>8</string>
             </property>
            </item>
           </widget>
           <widget class="QLabel" name="label_3">
            <property name="geometry">
             <rect>
              <x>130</x>
              <y>40</y>
              <width>42</width>
              <height>16</height>
             </rect>
            </property>
            <property name="text">
             <string>数据位:</string>
            </property>
           </widget>
           <widget class="QPushButton" name="openSerial">
            <property name="geometry">
             <rect>
              <x>90</x>
              <y>100</y>
              <width>71</width>
              <height>31</height>
             </rect>
            </property>
            <property name="text">
             <string>打开串口</string>
            </property>
           </widget>
           <widget class="QPushButton" name="updateSerialButton">
            <property name="geometry">
             <rect>
              <x>170</x>
              <y>10</y>
              <width>71</width>
              <height>23</height>
             </rect>
            </property>
            <property name="text">
             <string>刷新</string>
            </property>
           </widget>
          </widget>
          <widget class="QGroupBox" name="groupBox_5">
           <property name="geometry">
            <rect>
             <x>30</x>
             <y>140</y>
             <width>241</width>
             <height>301</height>
            </rect>
           </property>
           <property name="title">
            <string>虚拟坐标</string>
           </property>
           <widget class="QWidget" name="widget" native="true">
            <property name="geometry">
             <rect>
              <x>0</x>
              <y>20</y>
              <width>261</width>
              <height>271</height>
             </rect>
            </property>
            <property name="mouseTracking">
             <bool>true</bool>
            </property>
            <property name="acceptDrops">
             <bool>true</bool>
            </property>
           </widget>
          </widget>
         </widget>
         <widget class="QWidget" name="tab_2">
          <attribute name="title">
           <string>摄像设置</string>
          </attribute>
         </widget>
        </widget>
        <widget class="QGroupBox" name="groupBox_6">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>490</y>
           <width>301</width>
           <height>251</height>
          </rect>
         </property>
         <property name="title">
          <string>操作台</string>
         </property>
         <widget class="QSpinBox" name="pulseSpinBox">
          <property name="geometry">
           <rect>
            <x>150</x>
            <y>80</y>
            <width>61</width>
            <height>21</height>
           </rect>
          </property>
          <property name="maximum">
           <number>99999</number>
          </property>
          <property name="value">
           <number>65535</number>
          </property>
         </widget>
         <widget class="QPushButton" name="ySubButton">
          <property name="geometry">
           <rect>
            <x>120</x>
            <y>140</y>
            <width>81</width>
            <height>31</height>
           </rect>
          </property>
          <property name="text">
           <string>y轴负向</string>
          </property>
         </widget>
         <widget class="QPushButton" name="xAddButton">
          <property name="geometry">
           <rect>
            <x>220</x>
            <y>90</y>
            <width>71</width>
            <height>31</height>
           </rect>
          </property>
          <property name="text">
           <string>x轴正向</string>
          </property>
         </widget>
         <widget class="QPushButton" name="yAddButton">
          <property name="geometry">
           <rect>
            <x>120</x>
            <y>30</y>
            <width>81</width>
            <height>31</height>
           </rect>
          </property>
          <property name="text">
           <string>y轴正向</string>
          </property>
         </widget>
         <widget class="QPushButton" name="xSubButton">
          <property name="geometry">
           <rect>
            <x>0</x>
            <y>90</y>
            <width>71</width>
            <height>31</height>
           </rect>
          </property>
          <property name="text">
           <string>x轴负向</string>
          </property>
         </widget>
         <widget class="QPushButton" name="setPulseButton">
          <property name="geometry">
           <rect>
            <x>80</x>
            <y>90</y>
            <width>61</width>
            <height>31</height>
           </rect>
          </property>
          <property name="text">
           <string>设置脉冲</string>
          </property>
         </widget>
         <widget class="QPushButton" name="slowStop">
          <property name="geometry">
           <rect>
            <x>0</x>
            <y>180</y>
            <width>71</width>
            <height>31</height>
           </rect>
          </property>
          <property name="text">
           <string>减速停止</string>
          </property>
         </widget>
         <widget class="QPushButton" name="nowStop">
          <property name="geometry">
           <rect>
            <x>70</x>
            <y>180</y>
            <width>71</width>
            <height>31</height>
           </rect>
          </property>
          <property name="text">
           <string>立即停止</string>
          </property>
         </widget>
         <widget class="QPushButton" name="homeButton">
          <property name="geometry">
           <rect>
            <x>210</x>
            <y>180</y>
            <width>81</width>
            <height>31</height>
           </rect>
          </property>
          <property name="text">
           <string>Home</string>
          </property>
         </widget>
         <widget class="QCheckBox" name="constantCheckBox">
          <property name="geometry">
           <rect>
            <x>150</x>
            <y>110</y>
            <width>71</width>
            <height>16</height>
           </rect>
          </property>
          <property name="text">
           <string>连续驱动</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
         <widget class="QPushButton" name="followPoint">
          <property name="geometry">
           <rect>
            <x>140</x>
            <y>180</y>
            <width>71</width>
            <height>31</height>
           </rect>
          </property>
          <property name="text">
           <string>坐标追踪</string>
          </property>
         </widget>
         <widget class="QLabel" name="showLabel">
          <property name="geometry">
           <rect>
            <x>0</x>
            <y>10</y>
            <width>301</width>
            <height>31</height>
           </rect>
          </property>
          <property name="text">
           <string>dataResponse:</string>
          </property>
         </widget>
        </widget>
       </widget>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1345</width>
     <height>23</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>文件</string>
    </property>
    <addaction name="saveAllImage"/>
   </widget>
   <widget class="QMenu" name="menu_2">
    <property name="title">
     <string>操作</string>
    </property>
    <widget class="QMenu" name="menu_4">
     <property name="title">
      <string>显示模式设置</string>
     </property>
     <addaction name="oneMulone"/>
     <addaction name="twoMultwo"/>
     <addaction name="threeMulthree"/>
     <addaction name="threeMulfour"/>
     <addaction name="fourMulfour"/>
    </widget>
    <addaction name="menu_4"/>
    <addaction name="motorSpeed"/>
    <addaction name="actionShowFunction"/>
   </widget>
   <widget class="QMenu" name="menu_3">
    <property name="title">
     <string>帮助</string>
    </property>
    <addaction name="aboutAction"/>
    <addaction name="debugAction"/>
   </widget>
   <addaction name="menu"/>
   <addaction name="menu_2"/>
   <addaction name="menu_3"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="aboutAction">
   <property name="text">
    <string>关于</string>
   </property>
  </action>
  <action name="debugAction">
   <property name="text">
    <string>开发者模式</string>
   </property>
  </action>
  <action name="oneMulone">
   <property name="text">
    <string>1*1模式</string>
   </property>
  </action>
  <action name="twoMultwo">
   <property name="text">
    <string>2*2模式</string>
   </property>
  </action>
  <action name="threeMulthree">
   <property name="text">
    <string>3*3模式</string>
   </property>
  </action>
  <action name="threeMulfour">
   <property name="text">
    <string>3*4模式</string>
   </property>
  </action>
  <action name="fourMulfour">
   <property name="text">
    <string>4*4模式</string>
   </property>
  </action>
  <action name="saveAllImage">
   <property name="text">
    <string>保存全部</string>
   </property>
  </action>
  <action name="motorSpeed">
   <property name="text">
    <string>电机速率设置</string>
   </property>
  </action>
  <action name="actionShowFunction">
   <property name="text">
    <string>操作台</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
